ecosystem/sep-0010.md

```diff
@@ -476,6 +485,33 @@ with less than any threshold may allow a third-party to authenticate.
 A **Server** that needs to support validating accounts that do not exist can require a signature of the master key of
 the account address for accounts that do not exist.
 
+### Verifying Contract Accounts
+
+A **Server** can verify a contract account if the contract has a `web_auth_verify` function that has a `require_auth` called on its own address.
+
+When requesting the challenge the **Client** must include the `auth` parameter containing a prebuilt and presigned Soroban Authorization Entry to authorize a call to the contracts `web_auth_verify` function.
+
+The Soroban Authorization Entries should contain at least one entry that has:
+1. Credentials set to a Soroban Address Credentials with:
+   1. Address set to the **Client Account**.
+   1. Nonce set to a random value.
+   1. Signature Expiration Ledger set to a value in the near future to define the length of time of validatity of the signature.
+   1. Signature set to the values the account requires to authenticate, as defined by the contract.
+1. Root invocation with:
+   1. Host Function set to `HOST_FUNCTION_TYPE_INVOKE_CONTRACT` with:
+      1. Contract address set to the **Client Account**.
```

Comment:
I'm a bit confused; is the idea here for the account contract to implement `web_auth_verify` function? It would be sufficient to call a well-known contract that has a function like
```rust
fn web_auth_verify(addr: Address) {
  addr.require_auth();
}
```
The upside is that most custom accounts would work fine without adding a special codepath. The downside is the additional risk of needing to deal with state archival, but OTOH the same is true for the account contract and its data deps as well (which is probably worth specifying here as well - simulation can imitate entry restoration, so that might be ok, but we need to spell that explicitly).

Reply:
The intent was to add a codepath so custom accounts have to choose to participate in SEP-10 and can choose to specify a specific signer. But that might be overkill. You're right, it could be some other arbitrary contract call that is being signed for, instead of requiring signature of that specific fn, and then most custom accounts could work without additional code. I'll make that modification.

Reply:
> The intent was to add a codepath so custom accounts have to choose to participate in SEP-10 and can choose to specify a specific signer. 

Yeah, I don't think that's a reasonable place to opt out as it's possible to come up with SEP-N that allows auth without opt-in. I think opt-in should happen at the wallet level.